spec/guide_collection_spec.rb

```diff
@@ -0,0 +1,97 @@
+require 'spec_helper'
+
+describe GuideCollection do
+
+  describe '#insert' do
+    let!(:id) { GuideCollection.insert({name: 'foo', language: 'haskell', exercises: []})[:id] }
+    let(:inserted) { GuideCollection.find(id) }
+
+    it { expect(id).to_not be nil }
+    it { expect(inserted).to_not be nil }
+    it { expect(inserted.raw).to_not be_empty }
+    it { expect(inserted.id).to eq id }
+    it { expect(inserted.name).to eq 'foo' }
+    it { expect(inserted.exercises).to eq [] }
+    it { expect(GuideCollection.count).to eq 1 }
+
+    it { expect(inserted.to_json).to eq %Q{{"beta":false,"learning":false,"original_id_format":"%05d","name":"foo","language":"haskell","exercises":[],"id":"#{id}"}} }
```

Comment:
Testing like this is a bit tedious, because the failure is hard to detect. In platform, I've made [a helper](https://github.com/mumuki/mumuki-platform/blob/master/spec/controllers/api_controller_spec.rb#L4) for this kind of tests:

```ruby
RSpec::Matchers.define :json_eq do |expected_json_hash|
  match do |actual_json|
    expected_json_hash.with_indifferent_access == ActiveSupport::JSON.decode(actual_json)
  end
end
```

and is used like:

```ruby
let(:expected_json) {
  { guides: [
    {
      id: 1,
      github_repository: 'flbulgarelli/mumuki-sample-exercises',
      name: 'guide_1',
      language: { id: 1, name: 'language_1', image_url: 'lang1.jpeg' },
      path: { id: 1, name: 'category_1' },
      position: 1,
      exercises: []
    }
  ]}
}

it { expect(response.body).to json_eq expected_json }
```